eldritch/events.py

```diff
@@ -3231,6 +3239,10 @@ def resolve(self, state):
       assert self.combat.is_done()
       idx = self.monsters.index(self.combat.monster)
       self.done[idx] = True
+      for idx, monster in enumerate(self.monsters):
+        # Maybe an item defeated these monsters
```

Comment:
If you use the flute to defeat the monster, but the monster is endless or a pinata, then this won't work the way you want it to. Maybe check that the place of the monster == the character's place instead.